src/schemas/movies.py

```
@@ -1 +1,132 @@
-# Write your code here
+from datetime import date, timedelta
+from typing import List, Optional
+
+from pydantic import BaseModel, ConfigDict, Field, field_validator
+
+from database.models import MovieStatusEnum
+
+
+class GenreBaseSchema(BaseModel):
+    id: int
+    name: str
+
+    model_config = ConfigDict(from_attributes=True)
+
+
+class ActorBaseSchema(BaseModel):
+    id: int
+    name: str
+
+    model_config = ConfigDict(from_attributes=True)
+
+
+class CountryBaseSchema(BaseModel):
+    id: int
+    code: str
+    name: Optional[str]
+
+    model_config = ConfigDict(from_attributes=True)
+
+
+class LanguageBaseSchema(BaseModel):
+    id: int
+    name: str
+
+    model_config = ConfigDict(from_attributes=True)
+
+
+class MovieErrorSchema(BaseModel):
+    detail: str
+
+
+class MovieShortSchema(BaseModel):
+    id: int
+    name: str
+    date: date
+    score: float
+    overview: str
+
+    model_config = ConfigDict(from_attributes=True)
+
+
+class MovieListSchema(BaseModel):
+    movies: List[MovieShortSchema]
+    prev_page: Optional[str]
+    next_page: Optional[str]
+    total_pages: int
+    total_items: int
+
+
+class MovieDetailSchema(MovieShortSchema):
+    status: MovieStatusEnum
+    budget: float
+    revenue: float
+    country: CountryBaseSchema
+    genres: List[GenreBaseSchema]
+    actors: List[ActorBaseSchema]
+    languages: List[LanguageBaseSchema]
+
+
+class MovieCreateBaseSchema(BaseModel):
+    model_config = ConfigDict(from_attributes=True)
+
+    name: str = Field(max_length=255)
+    date: date
+    score: float = Field(ge=0, le=100)
+    overview: str
+    status: MovieStatusEnum
+    budget: float = Field(ge=0)
+    revenue: float = Field(ge=0)
+
+    @field_validator("date")
+    def validate_date(cls, value: date) -> date:
+        if value > date.today() + timedelta(days=365):
+            raise ValueError("The date must not be more than one year in the future.")
+        return value
```

The `date` validator on `MovieCreateBaseSchema` raises a specific error message: `"The date must not be more than one year in the future."`. The project requirements and tests expect invalid input to result in a `400` with the exact body `{"detail": "Invalid input data."` (this mapping is typically implemented in the route layer). To avoid leaking inconsistent messages from schema validators you can either: 1) raise a generic `ValueError('Invalid input data.')` here, or 2) keep this specific check but ensure the route-level exception handler maps all validation errors to the required 400 response and message. Decide and make it consistent.